app/components/post_draft/index.js

```diff
@@ -87,6 +106,9 @@ export function mapStateToProps(state, ownProps) {
         useChannelMentions,
         userIsOutOfOffice,
         value: currentDraft.draft,
+        groupsWithAllowReference: getAssociatedGroupsForReferenceMap(state, channelTeamId, channelId),
```

Comment:
Shoot! Good call on this ... I did not see this. It further made me realise that I should probably move this within the the `Min server = 5.24 && has license` if statement because if we don't have these, it makes no sense to try to retrieve groups with allow reference. 

Furthermore, I realised that in the `sendMessage` function, I can turn
`const groupMentions = this.groupsMentionedInText(value)` => `const groupMentions = notificationsToGroups ? this.groupsMentionedInText(value) : [];`.